### PR TITLE
dockerfile: new reference location

### PR DESCRIPTION
**Pull request by crazy-max**
~needs https://github.com/moby/buildkit/pull/2902~

Dockerfile reference is now located in the BuildKit repository.

Preview: https://deploy-preview-14948--docsdocker.netlify.app/engine/reference/builder/

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -160,3 +159,10 @@ fetch-remote:
       - dest: "registry"
         src:
           - "docs/configuration.md"
+
+  - repo: "https://github.com/crazy-max/buildkit" # FIXME: Use https://github.com/moby/buildkit
+    ref: "dockerfile-ref" # FIXME: Use master
```
Comment (crazy-max) [inline _config.yml:164]: using my fork and keeping in draft for now while waiting for input of @thaJeztah.

File: _config.yml [hunk 2]
```diff
@@ -116,8 +116,8 @@ defaults:
 #   - dest is the destination path within the working tree
 #   - src is a list of glob source paths within the remote repository
 fetch-remote:
-  - repo: "https://github.com/docker/cli"
-    ref: "20.10"
+  - repo: "https://github.com/crazy-max/docker-cli" # FIXME: Use https://github.com/docker/cli
+    ref: "20.10-dockerfile-ref-moved" # FIXME: Use 20.10
```
Comment (crazy-max) [inline _config.yml:120]: I have started to change links to the new location in my fork https://github.com/docker/cli/compare/20.10...crazy-max:20.10-dockerfile-ref-moved

One missing change in distribution repo to do:

```
 #16 173.1 - ./_site/registry/spec/deprecated-schema-v1/index.html
#16 173.1   *  linking to internal hash #from that does not exist (line 141)
#16 173.1      <a href="https://docs.docker.com/engine/reference/builder/#from">Dockerfile reference</a>
```

File: test.md [hunk 3]
```diff
@@ -107,7 +107,7 @@ culpa qui officia deserunt mollit anim id est laborum.
   - You can specify `org=foo` to use a Github organization other than Docker
 
 - A link to an auto-generated reference page that we pull in during docs builds:
-[/engine/reference/builder/#env](/engine/reference/builder/#env).
+[/dockerfile/reference/#env](/dockerfile/reference/#env).
```
Comment (thaJeztah) [inline test.md:110]: Perhaps we should start to consider using `/reference/xxxx` again for URLs instead;

- `/reference/dockerfile/`
- `/reference/commandline/xxx` etc

Comment (crazy-max) [reply test.md:110]: Hum indeed good point

Comment (crazy-max) [reply test.md]: Keeping in `/engine/reference/builder/` for now. Let's do the slug change in a follow-up.

**Comment by crazy-max** [comment]
@thaJeztah As discussed on Slack, we don't fetch from a release branch but master branch of the BuildKit repository for reference docs. Will open a follow-up PR on BuildKit to mark new features as suggested.

I have started to work on the Dockerfile reference changelog that will be on this repo in a dedicated page so we can cross-link with sections in reference docs but we need to first update the slug as the Dockerfile reference is not a section atm: `/engine/reference/builder`. Like you said we could switch to `/reference/dockerfile/` to access reference docs and in the same spirit have `/reference/dockerfile/release-notes` for dockerfile changelog. WDYT @thaJeztah?

**Comment by crazy-max** [comment]
Simplified this PR now that https://github.com/docker/docker.github.io/pull/14799 is merged. No need extra plugin to move pages around as the correct reference is picked up by fetch remote plugin to create edit and issue link: https://deploy-preview-14948--docsdocker.netlify.app/engine/reference/builder/

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**